pkg/cache/fakes/cache.go

```diff
@@ -83,13 +89,33 @@ func (f *FakeCache) FetchDataByIdentityWasNotCalled() bool {
 }
 
 func (f *FakeCache) FetchDataByDigest(ctx context.Context, name, digest string) (io.ReadCloser, error) {
+	if _, ok := f.fetchDataByDigestReturns[f.fetchDataByDigestCallCount]; !ok {
```

Comment:
Enhanced the cache to provide an error if return calls aren't configured...

Should probably apply it for all of them. :D But I thought we would edit it a need basis. :D